### ISSUE TITLE
Crash when one monitor is in power save

### ISSUE THREAD

**Issue by dnkl** [ifreund/waylock#55]
In a multi-monitor setup, if at least one monitor is in power saving mode (`wlopm --off`), waylock crashes when I try to unlock the session.

```
thread 21738 panic: reached unreachable code
/opt/zig-linux-x86_64-0.10.0/lib/std/debug.zig:278:14: 0x21e50a in assert (waylock)
/tmp/waylock/src/Lock.zig:382:11: 0x261f01 in submit_password (waylock)
/tmp/waylock/src/Seat.zig:160:41: 0x2589cd in keyboard_listener (waylock)
/tmp/waylock/zig-cache/zig-wayland/common.zig:107:69: 0x259289 in dispatcher (waylock)
error: failed to read password from pipe: EndOfStream
```

`WAYLAND_DEBUG=1`: [waylock.log](https://github.com/ifreund/waylock/files/10438284/waylock.log)

River version: https://github.com/riverwm/river/commit/8a3530b8a3d75aaa294d5cca718fc955d409f598
Waylock version: https://github.com/ifreund/waylock/commit/71e9cdb656839558d7221a973246c6f874f30167

**Comment by ifreund** [comment]
Hey, thanks for the bug report. The root cause here appears to actually be a bug in river that causes the locked event to never be sent due to a missing check for disabled outputs when deciding if all outputs have all their normal content hidden.

This river bug has exposed a latent waylock bug however as there shouldn't be an assertion failure if the user somehow manages to submit their password in the few milliseconds before the locked event is received.

**Comment by ifreund** [comment]
The river bug should now be fixed by https://github.com/riverwm/river/commit/6c7586e8d7c8c6a59b73864fc3096abf3f60d0ac. I'll have a look at fixing the minor waylock bug as well shortly though that one should be near impossible to hit in practice.

**Comment by dnkl** [comment]
Verified, thanks!

**Comment by ifreund** [comment]
Thanks for testing!

The possible waylock assertion failure should be fixed as well by 75b8dbec79c3276a6a95b8581ae463e61a919be8 :)
